src/Bicep.LangServer/Handlers/BicepTextDocumentSyncHandler.cs

```diff
@@ -18,10 +24,18 @@ namespace Bicep.LanguageServer.Handlers
     public class BicepTextDocumentSyncHandler : TextDocumentSyncHandlerBase
     {
         private readonly ICompilationManager compilationManager;
+        private readonly IConfigurationManager configurationManager;
+        private readonly ILinterRulesProvider linterRulesProvider;
+        private readonly ITelemetryProvider telemetryProvider;
 
-        public BicepTextDocumentSyncHandler(ICompilationManager compilationManager)
+        private readonly ConcurrentDictionary<DocumentUri, RootConfiguration> activeBicepConfigCache = new ConcurrentDictionary<DocumentUri, RootConfiguration>();
```

Comment:
>activeBicepConfigCache [](http://example.com/codeflow?start=78&length=22)

I see the code that keeps this dictionary up to date, but I don't see anything using the information inside. Did I miss something?

Reply:
On save, we want to send telemetry event. We can either do a TryGetValue(..) and then TryRemove(..) to keep the cache updated or TryRemove(..).

Let me know if I am missing something.

Reply:
Added some comments:
https://github.com/Azure/bicep/commit/1010ff5cc71629fccd7c55428ea3d10a167c03e1